integration/network/delete_test.go

```diff
@@ -0,0 +1,80 @@
+package network
+
+import (
+	"context"
+	"testing"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/integration/util/request"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func containsNetwork(nws []types.NetworkResource, nw types.NetworkCreateResponse) bool {
+	for _, n := range nws {
+		if n.ID == nw.ID {
+			return true
+		}
+	}
+	return false
+}
+
+// TestDockerNetworkDeletePreferID tests that if a network with a name
+// equal to another network's ID exists, the Network with the given
+// ID is removed, and not the network with the given name.
+func TestDockerNetworkDeletePreferID(t *testing.T) {
+	defer setupTest(t)()
+	client := request.NewAPIClient(t)
+	ctx := context.Background()
+
+	n1, err := client.NetworkCreate(ctx, "testNW", types.NetworkCreate{})
+	require.NoError(t, err)
+	n2, err := client.NetworkCreate(ctx, n1.ID, types.NetworkCreate{})
+	require.NoError(t, err)
+
+	nws, err := client.NetworkList(ctx, types.NetworkListOptions{})
+	require.NoError(t, err)
+
+	assert.Equal(t, true, containsNetwork(nws, n1), "Network n1 not found")
+	assert.Equal(t, true, containsNetwork(nws, n2), "Network n2 not found")
```

Comment:
Since these are just "sanity checks" that the setup was done correctly maybe this whole block (create networks, list and assert) can be moved out into a `setupOverlappingNetworks()` so that the test setup doesn't distract from the actual test case.